src/Framework/Framework/Resources/Scripts/binding-handlers/timer.ts

```diff
@@ -0,0 +1,48 @@
+﻿type TimerProps = {
+    interval: number,
+    enabled: KnockoutObservable<boolean>,
+    command: () => Promise<DotvvmAfterPostBackEventArgs>
+}
+
+ko.virtualElements.allowedBindings["dotvvm-timer"] = true;
+
+export default {
+    "dotvvm-timer": {
+        init: (element: HTMLElement, valueAccessor: () => TimerProps) => {
+            const prop = valueAccessor();
+            let timer: number | null = null;
+
+            if (ko.isObservable(prop.enabled)) {
```

Comment:
`enabled` can change even if the value isn't observable. For instance, it will happen if you set it to `Enabled={value: State == 2}`. There are two options to solve it

* Use the `binding.WrappedKnockoutExpression` property for the Enabled property in the DotvvmControl
* Add `update` handler
* Remove the condition here, and use `ko.pureComputed(() => ko.unwrap(valueAccessor().enabled).subscribe(...)`

The ko.pureComputed option is probably the easiest.

Reply:
Also, the subscription needs to be disposed. In this case, we cancel the timer, but it would get re-enabled if control is removed, and then Enabled changes.